From 707486a024859a14364f478cf24ad77391223fbb Mon Sep 17 00:00:00 2001
From: sandra <sandra@138bc75d-0d04-0410-961f-82ee72b054a4>
Date: Wed, 5 Aug 2009 00:46:07 +0000
Subject: [PATCH] 2009-08-04  Sandra Loosemore  <sandra@codesourcery.com>

	gcc/
	* doc/invoke.texi (MIPS Options): Document new 1004K -march options.
	* config/mips/mips.c (mips_cpu_info_table): Add 1004K cores.
	* config/mips/mips.h (MIPS_ISA_LEVEL_SPEC): Add pattern for 1004K.
	(MIPS_ARCH_FLOAT_SPEC): Likewise.
	(BASE_DRIVER_SELF_SPECS): Likewise.



git-svn-id: svn+ssh://gcc.gnu.org/svn/gcc/trunk@150475 138bc75d-0d04-0410-961f-82ee72b054a4
---
 gcc/ChangeLog          | 8 ++++++++
 gcc/config/mips/mips.c | 5 +++++
 gcc/config/mips/mips.h | 7 ++++---
 gcc/doc/invoke.texi    | 1 +
 4 files changed, 18 insertions(+), 3 deletions(-)

diff --git a/gcc/ChangeLog b/gcc/ChangeLog
index 1f1f4ae..8d5a293 100644
--- a/gcc/ChangeLog
+++ b/gcc/ChangeLog
@@ -3628,6 +3628,14 @@
 	make sure canon_rtx (mem_address) isn't simpler than
 	canon_rtx (expanded_mem_address).
 
+2009-08-04  Sandra Loosemore  <sandra@codesourcery.com>
+
+	* doc/invoke.texi (MIPS Options): Document new 1004K -march options.
+	* config/mips/mips.c (mips_cpu_info_table): Add 1004K cores.
+	* config/mips/mips.h (MIPS_ISA_LEVEL_SPEC): Add pattern for 1004K.
+	(MIPS_ARCH_FLOAT_SPEC): Likewise.
+	(BASE_DRIVER_SELF_SPECS): Likewise.
+
 2009-08-03  Janis Johnson  <janis187@us.ibm.com>
 
 	PR c/39902
diff --git a/gcc/config/mips/mips.c b/gcc/config/mips/mips.c
index 083b253..658d2da 100644
--- a/gcc/config/mips/mips.c
+++ b/gcc/config/mips/mips.c
@@ -659,6 +659,11 @@ static const struct mips_cpu_info mips_cpu_info_table[] = {
   { "74kx", PROCESSOR_74KF1_1, 33, 0 },
   { "74kf3_2", PROCESSOR_74KF3_2, 33, 0 },
 
+  { "1004kc", PROCESSOR_24KC, 33, 0 }, /* 1004K with MT/DSP.  */
+  { "1004kf2_1", PROCESSOR_24KF2_1, 33, 0 },
+  { "1004kf", PROCESSOR_24KF2_1, 33, 0 },
+  { "1004kf1_1", PROCESSOR_24KF1_1, 33, 0 },
+
   /* MIPS64 processors.  */
   { "5kc", PROCESSOR_5KC, 64, 0 },
   { "5kf", PROCESSOR_5KF, 64, 0 },
diff --git a/gcc/config/mips/mips.h b/gcc/config/mips/mips.h
index 8515ce1..e28e4c1 100644
--- a/gcc/config/mips/mips.h
+++ b/gcc/config/mips/mips.h
@@ -708,7 +708,7 @@ enum mips_code_readable_setting {
        |march=r10000|march=r12000|march=r14000|march=r16000:-mips4} \
      %{march=mips32|march=4kc|march=4km|march=4kp|march=4ksc:-mips32} \
      %{march=mips32r2|march=m4k|march=4ke*|march=4ksd|march=24k* \
-       |march=34k*|march=74k*: -mips32r2} \
+       |march=34k*|march=74k*|march=1004k*: -mips32r2} \
      %{march=mips64|march=5k*|march=20k*|march=sb1*|march=sr71000: -mips64} \
      %{march=mips64r2|march=octeon: -mips64r2} \
      %{!march=*: -" MULTILIB_ISA_DEFAULT "}}"
@@ -720,7 +720,8 @@ enum mips_code_readable_setting {
 #define MIPS_ARCH_FLOAT_SPEC \
   "%{mhard-float|msoft-float|march=mips*:; \
      march=vr41*|march=m4k|march=4k*|march=24kc|march=24kec \
-     |march=34kc|march=74kc|march=5kc|march=octeon: -msoft-float; \
+     |march=34kc|march=74kc|march=1004kc|march=5kc \
+     |march=octeon: -msoft-float;		  \
      march=*: -mhard-float}"
 
 /* A spec condition that matches 32-bit options.  It only works if
@@ -750,7 +751,7 @@ enum mips_code_readable_setting {
 
 /* A spec that infers the -mdsp setting from an -march argument.  */
 #define BASE_DRIVER_SELF_SPECS \
-  "%{!mno-dsp:%{march=24ke*|march=34k*|march=74k*: -mdsp}}"
+  "%{!mno-dsp:%{march=24ke*|march=34k*|march=74k*|march=1004k*: -mdsp}}"
 
 #define DRIVER_SELF_SPECS BASE_DRIVER_SELF_SPECS
 
diff --git a/gcc/doc/invoke.texi b/gcc/doc/invoke.texi
index a4bb385..cd0e0c2 100644
--- a/gcc/doc/invoke.texi
+++ b/gcc/doc/invoke.texi
@@ -12599,6 +12599,7 @@ The processor names are:
 @samp{24kec}, @samp{24kef2_1}, @samp{24kef1_1},
 @samp{34kc}, @samp{34kf2_1}, @samp{34kf1_1},
 @samp{74kc}, @samp{74kf2_1}, @samp{74kf1_1}, @samp{74kf3_2},
+@samp{1004kc}, @samp{1004kf2_1}, @samp{1004kf1_1},
 @samp{loongson2e}, @samp{loongson2f},
 @samp{m4k},
 @samp{octeon},
-- 
1.9.4

